next-app/app/schedule/page.tsx

```diff
@@ -0,0 +1,195 @@
+    </div>
+  );
+}
+
+function ScheduleData({ schedule, matches, unschedule }) {
+  const reformat = reformatMatches(matches);
+  const revSchedule = schedule.toReversed();
+
+  if (revSchedule.length > 0) {
+    return (
+      <div className="container pb-5">
+        <Accordion defaultActiveKey={revSchedule[0].id.toString()}>
+          {revSchedule.map((schedule) => {
+            const { id } = schedule;
+            const scheddate = new Date(
+              schedule.sdate.slice(0, 10) + 'T00:00:00.000-05:00',
+            ).toDateString();
+
+            return (
+              <div key={id}>
+                <Accordion.Item eventKey={id.toString()}>
+                  <Accordion.Header>{scheddate}</Accordion.Header>
+                  <Accordion.Body>
+                    <div>{matchesTable(reformat, scheddate)}</div>
+                    <div className="text-end">
+                      <ConfirmationModal
+                        unscheduleID={() => unschedule(id)}
+                        unscheduledDate={scheddate}
+                      />
+                    </div>
+                  </Accordion.Body>
+                </Accordion.Item>
+              </div>
+            );
+          })}
+        </Accordion>
+      </div>
+    );
+  } else {
+    return (
+      <div className="text-center">
+        <p>No scheduled dates found</p>
+      </div>
+    );
+  }
+}
+
+function reformatMatches(matches) {
+  const reformat = matches.map((match) => {
+    const { teamA, teamB, ...rest } = match;
+    const a = match.teamA.map((a) => {
+      return a.pname;
+    });
+    const b = match.teamB.map((b) => {
+      return b.pname;
+    });
+    rest.teamA = a;
+    rest.teamB = b;
+    return rest;
+  });
+  return reformat;
+}
+
+function matchesTable(matchesArr, scheduleDate) {
+  let matchesTable = [];
+  for (let i = 0; i < matchesArr.length; i++) {
+    if (matchesArr[i].mdate === scheduleDate) {
+      matchesTable.push(matchesArr[i]);
+    }
+  }
+  if (matchesTable.length > 0) {
+    return (
+      <table className="table text-center">
+        <thead>
+          <tr>
+            <th>#</th>
+            <th className="text-primary">Team A</th>
+            <th>Score</th>
+            <th className="text-danger">Team B</th>
+          </tr>
+        </thead>
+        <tbody>
+          {matchesTable.map((matches) => {
+            const { match_id } = matches;
+            return (
+              <tr key={match_id}>
+                <td>{match_id}</td>
+                <td>{matches.teamA.join(', ')}</td>
+                <td>
+                  {matches.ascore} &ndash; {matches.bscore}
+                </td>
+                <td>{matches.teamB.join(', ')}</td>
+              </tr>
+            );
+          })}
+        </tbody>
+      </table>
+    );
+  } else {
+    return <p>No match data found.</p>;
+  }
+}
+
+function ConfirmationModal({ unscheduleID, unscheduledDate }) {
```

Comment:
may want the name to be descriptive in a way that implies the button is a part of this component. that or take the button out of the component.